napari/layers/image/image.py

```diff
@@ -388,21 +388,20 @@ def _new_empty_slice(self):
     def _get_empty_image(self):
         """Get empty image to use as the default before data is loaded."""
         if self.rgb:
-            return np.zeros((1,) * self._ndisplay + (3,))
+            return np.zeros((1,) * self._slice_input.ndisplay + (3,))
         else:
-            return np.zeros((1,) * self._ndisplay)
+            return np.zeros((1,) * self._slice_input.ndisplay)
 
-    def _get_order(self):
+    def _get_order(self) -> Tuple[int]:
         """Return the order of the displayed dimensions."""
+        displayed_order = tuple(np.argsort(self._slice_input.displayed))
```

Comment:
There were several with this code + comment 
```
# equivalent to: order = np.argsort(displayed)
order = sorted(range(len(displayed)), key=lambda x: displayed[x])
````

I think it was for performance reasons. Is it worth considering here as well?

Reply:
Maybe, but keep in mind we're always sorting 2 or 3 numbers. I can write some really fast sorting algorithms for that case :p